src/oidcop/oauth2/token.py

```diff
@@ -358,6 +363,289 @@ def post_parse_request(
+                error_description="Unauthorized scope requested",
+            )
+
+        subject_token_type = request["subject_token_type"]
+        if subject_token_type not in config["policy"]:
+            if "" not in config["policy"]:
+                raise ImproperlyConfigured(
+                    "subject_token_type {subject_token_type} missing from "
+                    "policy and no default is defined"
+                )
+            subject_token_type = ""
+
+        policy = config["policy"][subject_token_type]
+        callable = policy["callable"]
+        kwargs = policy["kwargs"]
+
+        try:
+            if isinstance(callable, str):
+                fn = importer(callable)
+            else:
+                fn = callable
+        except Exception:
+            raise ImproperlyConfigured(
+                "Error importing {callable} policy callable"
+            )
+
+        try:
+            return fn(request, context=_context, subject_token=token, **kwargs)
+        except Exception:
+            logger.error("Error while executing the {fn} policy callable")
+            return self.error_cls(
+                error="server_error",
+                error_description="Internal server error"
+            )
+
+    def token_exchange_response(self, token):
+        response_args = {}
+        response_args["access_token"] = token.value
+        response_args["scope"] = token.scope
+        response_args["issued_token_type"] = token.token_class
+        response_args["expires_in"] = token.usage_rules.get("expires_in", 0)
+        if hasattr(token, "token_type"):
+            response_args["token_type"] = token.token_type
+        else:
+            response_args["token_type"] = "N_A"
+
+        return TokenExchangeResponse(**response_args)
+
+    def process_request(self, request, **kwargs):
+        _context = self.endpoint.server_get("endpoint_context")
+        _mngr = _context.session_manager
+        try:
+            _session_info = _mngr.get_session_info_by_token(
+                request["subject_token"], grant=True
+            )
+        except ToOld:
+            logger.error("Subject token has expired.")
+            return self.error_cls(
+                error="invalid_request",
+                error_description="Subject token has expired"
+            )
+        except (KeyError, UnknownToken):
+            logger.error("Subject token invalid.")
+            return self.error_cls(
+                error="invalid_request",
+                error_description="Subject token invalid"
+            )
+
+        token = _mngr.find_token(_session_info["session_id"], request["subject_token"])
+        _requested_token_type = request.get("requested_token_type",
+                                           "urn:ietf:params:oauth:token-type:access_token")
+
+        _token_class = _requested_token_type.split(":")[-1]
+        if _token_class == "access_token":
+            _token_type = _token_class
```

Comment:
This must be `Bearer` or `DPoP`. We could use the same logic as in https://github.com/IdentityPython/oidc-op/pull/165/files#diff-701a8905a1282dde6d1b7325ab65768a840a3942e41112d24135688431564df3R255-R262 ?